daemon/snap.go

```diff
@@ -111,6 +111,13 @@ func allLocalSnapInfos(st *state.State) ([]aboutSnap, error) {
 	return about, firstErr
 }
 
+func effectiveConfinement(snapst *snapstate.SnapState) snap.ConfinementType {
```

Comment:
If we are aggregating, we should probably do on the opposite end: a DevMode bool flag on client.Snap which is set to true when we see a confinement=devmode snap from the store. This feels simpler and doesn't blur the line of what the "confinement" property means everywhere else (a statement by the snap itself of how it would like to be handled, rather than current status).